File: src/org/magnum/logger/connectivity/usb/UsbTable.java
package org.magnum.logger.connectivity.usb;

import org.magnum.logger.Database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

public class UsbTable extends Database
{

	private static final String TABLE_NAME = "Bluetooth";
	private static final String COLUMN_NAMES[] = { "Time", "Name", "Class",  "Change" };

	public UsbTable(Context context)
	{
		super(context);
	}

	public static void create(SQLiteDatabase db)
	{
		db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + "( Time INTEGER NOT NULL,  Name TEXT NOT NULL, Class TEXT NOT NULL, Change TEXT NOT NULL )");
	}

	@Override
	public String[] getColumnNames()
	{
		return COLUMN_NAMES;
	}

	@Override
	public String getTableName()
	{
		return TABLE_NAME;
	}
}

File: app/src/main/java/org/magnum/logger/connectivity/usb/UsbTable.kt
package org.magnum.logger.connectivity.usb

import androidx.room.*

const val NAME = "Usb"

@Entity(tableName = NAME)
data class UsbRecord(@PrimaryKey val time: Long, val name: String, @ColumnInfo(name = "class") val clazz: String, val change: String)

@Dao
interface UsbTable {
    @Insert
    fun insert(vararg record: UsbRecord)
}